### DISCUSSION TITLE
1.0.0

### DISCUSSION THREAD

**Discussion by patrick-zippenfenig** [open-meteo/open-meteo#587] [Announcements]
🎉🎉🎉 Exciting News! The Open-Meteo source code has officially reached version 1.0.0, marking a significant milestone. This release introduces a stable distribution for both Docker and Ubuntu packages, ensuring a more seamless and reliable deployment experience! 🎉🎉🎉

Major changes include a restructuring of the internal database layout and the incorporation of new weather models. The redesigned database layout is geared towards long-term stability, ensuring compatibility with future versions.

‼️ Important Notice: For users upgrading from previous versions utilizing Docker or Ubuntu Prebuilt packages, it is important to execute a database migration script to preserve previously downloaded weather data. Additionally, many weather models and variables have undergone renaming for consistency. Refer to [this PR](https://github.com/open-meteo/open-meteo/pull/543) for detailed information. ‼️

Version 1.0.0 also establishes the groundwork for redistributing the Open-Meteo database as open-data through an AWS Open-Data Sponsorship. Instead of directly downloading raw weather data in GRIB format from national weather services, users can now access the optimized time-series database from Open-Meteo directly through AWS. The open-data distribution allows fine control, enabling users to retrieve only a specific subset of weather variables relevant to their use case. This not only conserves resources for data processing but also reduces bandwidth usage on the servers of national weather services. Instructions can be found in the [getting started guide](/docs/getting-started.md) and the [open-data documentation](https://github.com/open-meteo/open-data). Please be aware that not all weather models and data sources are currently available and will be progressively added over the next few weeks with documentation and examples to follow.

We sincerely appreciate your continuous support and trust that you'll find immense value in the latest enhancements introduced to the Open-Meteo weather API!

## What's Changed
* Import FlatBuffers generated files from client repository by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/461
* fix: ICON-D2 EPS download for rain/showers and upper level wind by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/465
* fix: bump github.com/vapor/vapor from 4.84.4 to 4.84.6 by @dependabot in https://github.com/open-meteo/open-meteo/pull/477
* fix: bump github.com/apple/swift-nio-http2 from 1.27.0 to 1.28.1 by @dependabot in https://github.com/open-meteo/open-meteo/pull/476
* fix: bump github.com/vapor/console-kit from 4.8.1 to 4.9.0 by @dependabot in https://github.com/open-meteo/open-meteo/pull/475
* fix: bump github.com/vapor/routing-kit from 4.8.0 to 4.8.1 by @dependabot in https://github.com/open-meteo/open-meteo/pull/474
* fix: bump SamKirkland/FTP-Deploy-Action from 4.3.0 to 4.3.4 by @dependabot in https://github.com/open-meteo/open-meteo/pull/473
* fix: bump docker/setup-qemu-action from 2 to 3 by @dependabot in https://github.com/open-meteo/open-meteo/pull/472
* fix: bump docker/login-action from 2 to 3 by @dependabot in https://github.com/open-meteo/open-meteo/pull/471
* fix: bump docker/setup-buildx-action from 2 to 3 by @dependabot in https://github.com/open-meteo/open-meteo/pull/470
* fix: bump docker/metadata-action from 4 to 5 by @dependabot in https://github.com/open-meteo/open-meteo/pull/469
* More efficient FlatBuffers encoding by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/481
* fix: bump docker/build-push-action from 4 to 5 by @dependabot in https://github.com/open-meteo/open-meteo/pull/478
* fix: bump github.com/apple/swift-nio from 2.60.0 to 2.61.0 by @dependabot in https://github.com/open-meteo/open-meteo/pull/489
* fix: bump github.com/vapor/async-kit from 1.18.0 to 1.19.0 by @dependabot in https://github.com/open-meteo/open-meteo/pull/493
* fix: bump github.com/vapor/console-kit from 4.9.0 to 4.9.1 by @dependabot in https://github.com/open-meteo/open-meteo/pull/495
* fix: bump github.com/vapor/vapor from 4.84.6 to 4.85.1 by @dependabot in https://github.com/open-meteo/open-meteo/pull/498
* fix: bump github.com/apple/swift-nio from 2.61.0 to 2.61.1 by @dependabot in https://github.com/open-meteo/open-meteo/pull/492
* Make timezones optional for daily data by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/496
* fix: bump github.com/open-meteo/sdk from 1.4.0 to 1.5.0 by @dependabot in https://github.com/open-meteo/open-meteo/pull/497
* fix: bump github.com/vapor/console-kit from 4.9.1 to 4.10.1 by @dependabot in https://github.com/open-meteo/open-meteo/pull/500
* fix: bump github.com/vapor/vapor from 4.85.1 to 4.86.0 by @dependabot in https://github.com/open-meteo/open-meteo/pull/499
* fix: bump github.com/vapor/vapor from 4.86.0 to 4.86.2 by @dependabot in https://github.com/open-meteo/open-meteo/pull/503
* fix: bump github.com/vapor/routing-kit from 4.8.1 to 4.8.2 by @dependabot in https://github.com/open-meteo/open-meteo/pull/502
* feat: Integrate daylight and sunshine duration by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/507
* fix: Missing ECMWF rain correctly label as NaN by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/508
* feat: Add ERA5 Ocean waves to Marine API with 80 years history by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/510
* fix: bump github.com/apple/swift-algorithms from 1.1.0 to 1.2.0 by @dependabot in https://github.com/open-meteo/open-meteo/pull/509
* Fix: Protect JMA downloader against overlapping runs by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/516
* Added WetBulb to the app list by @Isma1306 in https://github.com/open-meteo/open-meteo/pull/520
* fix: bump github.com/vapor/vapor from 4.86.2 to 4.87.1 by @dependabot in https://github.com/open-meteo/open-meteo/pull/517
* feat: Output location_id in JSON format if not 0 by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/522
* fix: Correct DWD ICON weather codes to remove drizzle if precipitation is 0 by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/523
* fix: approximate freezing level height for grid-elevation below ground by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/524
* fix: CORS preflight support by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/528
* feat: Select number of hourly time steps independently from daily by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/530
* Fix: Define liquid rain in meteofrance models by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/531
* feat: Integrate additional layers for GFS and HRRR by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/532
* Domain registry and refactoring by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/538
* Support POST requests for longer locations lists by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/544
* Fix negative snow depth by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/546
* BREAKING CHANGE: New directory structure for local database files by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/543
* fix: bump github.com/vapor/vapor from 4.89.0 to 4.89.1 by @dependabot in https://github.com/open-meteo/open-meteo/pull/547
* Rework NCEP GFS downloader by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/551
* fix: bump github.com/vapor/vapor from 4.89.1 to 4.89.3 by @dependabot in https://github.com/open-meteo/open-meteo/pull/550
* fix: bump actions/upload-artifact from 3 to 4 by @dependabot in https://github.com/open-meteo/open-meteo/pull/553
* Use new MeteoFrance API by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/561
* Use `sendable` for life cycle objects and update all controllers by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/568
* fix: bump github.com/swift-server/async-http-client from 1.19.0 to 1.20.0 by @dependabot in https://github.com/open-meteo/open-meteo/pull/565
* fix: bump github.com/vapor/routing-kit from 4.8.2 to 4.9.0 by @dependabot in https://github.com/open-meteo/open-meteo/pull/564
* fix: bump github.com/apple/swift-collections from 1.0.5 to 1.0.6 by @dependabot in https://github.com/open-meteo/open-meteo/pull/563
* fix: bump github.com/vapor/multipart-kit from 4.5.4 to 4.6.0 by @dependabot in https://github.com/open-meteo/open-meteo/pull/557
* CMA GFS GRAPES model integration by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/573
* Integrate Australian Bureau of Meteorology (BOM) forecasts by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/575
* fix: bump github.com/vapor/vapor from 4.90.0 to 4.91.1 by @dependabot in https://github.com/open-meteo/open-meteo/pull/582
* fix: bump github.com/vapor/console-kit from 4.14.0 to 4.14.1 by @dependabot in https://github.com/open-meteo/open-meteo/pull/586
* fix: bump github.com/apple/swift-crypto from 3.1.0 to 3.2.0 by @dependabot in https://github.com/open-meteo/open-meteo/pull/584
* Update getting started by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/pull/560

## New Contributors
* @Isma1306 made their first contribution in https://github.com/open-meteo/open-meteo/pull/520

**Full Changelog**: https://github.com/open-meteo/open-meteo/compare/0.2.89...1.0.0

About the [Open-Meteo Weather API](https://open-meteo.com): Open-Meteo is a weather API that operates on an open-source basis and provides free access for non-commercial purposes. No API key is necessary; you can start using it right away!

At Open-Meteo, we firmly believe that access to accurate and dependable weather data should be accessible to all. That's why we've developed an open-source weather API that utilizes weather forecasts derived from open-data sources offered by national weather services. Unlike other weather APIs, Open-Meteo grants complete access to its source code, and all data sources are openly acknowledged, giving credit to the national weather services for their valuable contributions. Users can set up their own weather API rapidly using Docker or prebuilt Ubuntu packages. By sharing the source code, users can thoroughly scrutinize the weather data processing and even make adjustments as needed. We actively encourage and welcome contributions from our user community.

This API is freely available for non-commercial use, with no associated costs. Despite being free of charge, it delivers top-tier forecast accuracy. Leveraging an extensive range of local weather models with frequent updates, the API ensures the generation of highly precise forecasts for locations worldwide.

We appreciate your consideration of Open-Meteo for your weather data requirements. We are continuously working to enhance our services and are open to any feedback or inquiries you may have. Please do not hesitate to reach out to us at [info@open-meteo.com](mailto:info@open-meteo.com).

<hr /><em>This discussion was created from the release <a href='https://github.com/open-meteo/open-meteo/releases/tag/1.0.0'>1.0.0</a>.</em>

**Comment by MantraMedia** [comment]
Wow! Congratulations! 🎉

**Comment by willp** [comment]
Congrats on 1.0.0 and on getting AWS open data status!

Upgrading from pre-1.0 to 1.0.0 with the migration subcommand worked perfectly for renaming all my database files, and it's nice that it requires the `--execute` flag to modify the filesystem, as a safety feature.  I was even able to set up some quick and dirty NFS mounts to my backup server(s) and ran the migration command in a separate directory, with a local `./data/` NFS mountpoint onto the remote backup.  I'm glad for the normalized naming conventions and the more hierarchical directory structure.

Thanks for all your hard work!!

**Reply by patrick-zippenfenig** [reply]
Glad to hear migration went well!

**Comment by ouardito** [comment]
Congratulatiosn on the new Version Achieved ! 

Quick question, any plans on c++ sdk ? or is it already in review/process ?

Thank you !

**Reply by patrick-zippenfenig** [reply]
Hi, no there are no plans for a c++ sdk/library. However, you can use the provided flatbuffers schema files and compile them to C++ as a starting point: https://github.com/open-meteo/sdk

**Comment by dzungpv** [comment]
Great news, I back to this after month running it on my server. I ask a question: Is full data available on the AWS? I will migrate to open data to reduce server load.
